### PR TITLE
[Calamari]Add calamari runtime, bump deps to polkadot-v0.9.8

### LINKED ISSUE #110: Calamari Shell Chain Tracking Issue
- [x] Calamari parachain runtime pallet configuration
- [x] bump dependency to newest (0.9.8)
- [x] Use cargo features to decide which runtime can be compiled
- [x] filter all functions
- [x] PR to polkadot-js for updating calamari icon and some naming stuff -> [PR](https://github.com/polkadot-js/apps/pull/5832)
- [x] Test runtime upgrade locally
- [ ] End to end test using `Manta-Subxt` (try the current code) (Good to have)
- [ ] Weight definitions and related settings. (Good to have)
- [ ] Test shell chain on EC2 (own-relay chain, 5 collators).
   - [x] EC2 deployment (with Rob) 
   - [x] producing blocks
   - [x] basic functionalities (filtering out functions)
   - [x] release workflow (binary and WASM) (Jamie, Shumo, Gerogi, Qiudong, Rob)
       - [x] first calamari release
       - [ ] automation
         - [ ] dockerize [Rob]
         - [x] srtool for wasm
       - [x] change log (generator [implemented](https://github.com/Manta-Network/Manta/compare/c4de44d..2f254db), needs customisation)
   - [x] shell chain genesis and secret configuration  
   - [x] Dry run of the crowd loan
   - [x] Integrating with Kevin/Sinan's frontend
   - [x]  test runtime upgrade (Qiudong)
   - [x] Show calamari network in dropdown menu from polkadot-js-extension (Georgi)
- [ ] Test shell chain on external testnet (using our own chain spec) (Jamie)
   - [ ] EC2 deployment (with Rob) 
   - [ ] producing blocks
   - [ ] basic functionalities (filtering out functions)
   - [ ] runtime upgrade 
   - [ ] stress tests (make sure our weight configuration is good enough to handle DDoS) (Good to have)
   - [ ] Dry run of the crowd loan
   - [ ] Integrating with Kevn/Sinan's frontend, create a hosted version (Kevin, Rob)
- [x] crowdloan front-end
   - [x] contribution history
   - [x] referral
   - [x] leaderboard
   - [x] trend
- [x] investigate the performance regression issue when illegal collators spawned (Shumo, Georgi) 
- [x] Configure Calamari's SS58 Prefix
- [ ] Configure Calamari runtime chain spec(session keys, root key, peer id, etc)
   - [x] figure out the semantics of the constants 
- [x] Impl `try-runtime` for Calamari runtime (Jamie, Shumo)
- [x] Calamari website/doc depolyment (Georgi, Qiudong)
- [x] Shell chain testnet genesis
- [x] Shell chain genesis

### PR DESCRIPTION
**Pull request by Dengjianping**
1. Add calamari runtime.
2. Bump deps to polkadot-v0.9.8(~~polkadot-v0.9.7~~).
3. Add call filter for calamari runtime.

Related commits:
~~polkadot: branch = "release-v0.9.7"~~
~~substrate: branch = "polkadot-v0.9.7"~~
~~cumulus: branch = "polkadot-v0.9.7"~~
polkadot: branch = "release-v0.9.8"
substrate: branch = "polkadot-v0.9.8"
cumulus: branch = "polkadot-v0.9.8"

### CODE REVIEW

**Comment by ghzlatarev** [comment]
> Closes #110
> 
> 1. Add calamari runtime.
> 2. Bump deps to polkadot-v0.9.7.
> 3. Add call filter for calamari runtime.

Can you include the polkadot, cumulus and substrate commits like the other PRs, please.

**Comment by Dengjianping** [comment]
> > Closes #110
> > 
> > 1. Add calamari runtime.
> > 2. Bump deps to polkadot-v0.9.7.
> > 3. Add call filter for calamari runtime.
> 
> Can you include the polkadot, cumulus and substrate branches like the other PRs, please.

Sorry for that, it's updated now.

File: runtime/calamari/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,890 @@
...
+				MAXIMUM_BLOCK_WEIGHT - NORMAL_DISPATCH_RATIO * MAXIMUM_BLOCK_WEIGHT
+			);
+		})
+		.avg_block_initialization(AVERAGE_ON_INITIALIZE_RATIO)
+		.build_or_panic();
+	pub const SS58Prefix: u8 = manta_primitives::constants::CALAMARI_SS58PREFIX;
+}
+
+// Don't allow permission-less asset creation.
+pub struct BaseFilter;
```
Comment (ghzlatarev) [inline runtime/calamari/src/lib.rs:136]: This is missing in manta-pc runtime, should we add it there as well ?

Comment (Dengjianping) [reply runtime/calamari/src/lib.rs:136]: No, this filter is only for calamari for now.

File: node/src/chain_spec.rs [hunk 2]
```diff
@@ -316,6 +336,270 @@ fn manta_pc_testnet_genesis(
...
+				],
+				id,
+			)
+		},
+		vec![],
+		None,
+		Some(CALAMARI_PROTOCOL_ID),
+		Some(properties),
+		Extensions {
+			relay_chain: "polkadot-dev".into(),
```
Comment (ghzlatarev) [inline node/src/chain_spec.rs]: Maybe we can make the chain spec strings constants, as well.

File: runtime/calamari/src/lib.rs (hunk 1 above)
Comment (ghzlatarev) [reply runtime/calamari/src/lib.rs:136]: Gotcha

File: node/src/chain_spec.rs (hunk 2 above)
Comment (Dengjianping) [reply node/src/chain_spec.rs]: Followed your comment.

**Review by ghzlatarev — APPROVED** [review]

**Comment by Dengjianping** [comment]
@stechu @ghzlatarev @xialovexia 
Please note, I bumped deps to polkadot-v0.9.8.

File: node/src/service.rs [hunk 3]
```diff
@@ -472,6 +478,8 @@ where
 						slot_duration,
 						// We got around 500ms for proposing
 						block_proposal_slot_portion: SlotProportion::new(1f32 / 24f32),
+						// And a maximum of 750ms if slots are skipped
+						max_block_proposal_slot_portion: Some(SlotProportion::new(1f32 / 16f32)),
```
Comment (stechu) [inline node/src/service.rs:483]: Please mark the origin of the code (e.g. statemint).

File: runtime/calamari/src/lib.rs [hunk 4]
```diff
@@ -0,0 +1,906 @@
+//! Manta Parachain runtime.
```
Comment (stechu) [inline runtime/calamari/src/lib.rs]: please add the template header here.

File: runtime/calamari/src/lib.rs [hunk 5]
```diff
@@ -0,0 +1,906 @@
...
+	type CallHasher = BlakeTwo256;
+	type AnnouncementDepositBase = AnnouncementDepositBase;
+	type AnnouncementDepositFactor = AnnouncementDepositFactor;
+}
+
+parameter_types! {
+	pub const MinVestedTransfer: Balance = 10 * MA;
+}
+
+impl pallet_vesting::Config for Runtime {
```
Comment (stechu) [inline runtime/calamari/src/lib.rs]: `pallet_vesting` doesn't seem necessary. especially initially.

File: runtime/calamari/src/lib.rs [hunk 6]
```diff
@@ -0,0 +1,906 @@
...
+	type DisabledValidatorsThreshold = DisabledValidatorsThreshold;
+	type WeightInfo = ();
+}
+
+impl pallet_aura::Config for Runtime {
+	type AuthorityId = AuraId;
+}
+
+parameter_types! {
+	pub const PotId: PalletId = PalletId(*b"PotStake");
```
Comment (stechu) [inline runtime/calamari/src/lib.rs:483]: Can you add some comments about the semantics of these constants.

File: runtime/calamari/src/lib.rs [hunk 7]
```diff
@@ -0,0 +1,906 @@
...
+		Block = Block,
+		NodeBlock = opaque::Block,
+		UncheckedExtrinsic = UncheckedExtrinsic,
+	{
+		// System support stuff.
+		System: frame_system::{Pallet, Call, Config, Storage, Event<T>} = 0,
+		ParachainSystem: cumulus_pallet_parachain_system::{Pallet, Call, Storage, Inherent, Event<T>} = 1,
+		Timestamp: pallet_timestamp::{Pallet, Call, Storage, Inherent} = 2,
+		ParachainInfo: parachain_info::{Pallet, Storage, Config} = 3,
+		Scheduler: pallet_scheduler::{Pallet, Call, Storage, Event<T>} = 4,
```
Comment (stechu) [inline runtime/calamari/src/lib.rs]: Please remove the scheduler pallet and test upgrade without scheduler.

File: runtime/manta-pc/src/lib.rs [hunk 8]
```diff
@@ -626,13 +634,11 @@ construct_runtime!(
 		// System support stuff.
 		System: frame_system::{Pallet, Call, Config, Storage, Event<T>} = 0,
 		ParachainSystem: cumulus_pallet_parachain_system::{Pallet, Call, Storage, Inherent, Event<T>} = 1,
-		RandomnessCollectiveFlip: pallet_randomness_collective_flip::{Pallet, Call, Storage} = 2,
+		RandomnessCollectiveFlip: pallet_randomness_collective_flip::{Pallet, Storage} = 2,
```
Comment (stechu) [inline runtime/manta-pc/src/lib.rs]: Please remove this pallet.

**Review by stechu — CHANGES_REQUESTED** [review]
Remaining issues:
1. need to switch binary names when using different features (or if this is not possible, then we can think how to make this more intuitive)
2. have a default setting
3. figure out what candidacy_bond is, and set the value properly
4. check whether we need pallet-proxy or not, if not remove the pallet from the configuration
5. fix all the inlined comments

File: runtime/manta-pc/src/lib.rs (hunk 8 above)
Comment (Dengjianping) [reply runtime/manta-pc/src/lib.rs]: Removed.

File: runtime/calamari/src/lib.rs (hunk 7 above)
Comment (Dengjianping) [reply runtime/calamari/src/lib.rs]: Removed, and @xialovexia did runtime upgrade, it worked by `set_code` directly.

File: runtime/calamari/src/lib.rs (hunk 6 above)
Comment (Dengjianping) [reply runtime/calamari/src/lib.rs:483]: Added.

File: runtime/calamari/src/lib.rs (hunk 5 above)
Comment (Dengjianping) [reply runtime/calamari/src/lib.rs]: Removed.

File: node/src/service.rs (hunk 3 above)
Comment (Dengjianping) [reply node/src/service.rs:483]: Added.

File: runtime/calamari/src/lib.rs (hunk 4 above)
Comment (Dengjianping) [reply runtime/calamari/src/lib.rs]: Updated.

**Review by stechu — APPROVED** [review]

**Pull request merged**